internals/daemon/api_notices.go

```diff
@@ -42,6 +45,38 @@ type addedNotice struct {
 func v1GetNotices(c *Command, r *http.Request, _ *UserState) Response {
 	query := r.URL.Query()
 
+	requestUID, err := uidFromRequest(r)
+	if err != nil {
+		return statusForbidden("cannot determine UID of request, so cannot retrieve notices")
+	}
+	// By default, return notices with the request UID and public notices.
+	userID := &requestUID
+	daemonUID := uint32(sysGetuid())
+
+	if len(query["user-id"]) > 0 {
+		if !isAdmin(requestUID, daemonUID) {
+			return statusForbidden(`only admins may use the "user-id" filter`)
+		}
+		userID, err = sanitizeUserIDFilter(query["user-id"])
+		if err != nil {
+			return statusBadRequest(`invalid "user-id" filter: %v`, err)
+		}
+	}
+
+	if len(query["select"]) > 0 {
+		if !isAdmin(requestUID, daemonUID) {
+			return statusForbidden(`only admins may use the "select" filter`)
+		}
+		if query["user-id"] != nil {
```

Comment:
Any reason not to use the same `len(x) > 0` test as above? (in case it's a zero-length slice).

```suggestion
		if len(query["user-id"]) > 0 {
```

Reply:
Nope, just an oversight by me. I couldn't remember at first whether we were checking for `nil`-ness or length, and didn't go back to fix this one after I switched to length elsewhere.